maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java

```diff
@@ -160,13 +187,113 @@ private boolean buildCoreIfWatched(Server server) throws MojoFailureException {
         return false;
     }
 
-    private void buildOwaProject() throws MojoExecutionException {
+    protected void buildOwaProject() throws MojoExecutionException {
+        wizard.showMessage("Building OWA project...");
+
+        boolean isUsingSystemNpmAndNodejs;
+
+        Map<String, String> systemNodeAndNpmVersion = getSystemNodeAndNpmVersion();
+        Map<String, String> batchNodeAndNpmVersion = getBatchNodeAndNpmVersion();
```

Comment:
Please get rid of the maps. I don't get why you do not simply use npmVersion and nodeVersion variables...